libres/tests/analysis/test_compare_initX.cpp

```diff
@@ -0,0 +1,181 @@
+#include <catch2/catch.hpp>
+#include <vector>
+#include <cmath>
+
+#include <ert/util/rng.h>
+#include <ert/enkf/enkf_util.hpp>
+
+#include <ert/util/bool_vector.hpp>
+#include <ert/enkf/obs_data.hpp>
+#include <ert/enkf/meas_data.hpp>
+#include <ert/analysis/ies/ies_enkf.hpp>
+#include <ert/analysis/std_enkf.hpp>
+
+struct model {
+    double amplitude;
+    double phase;
+
+    model(double a, double p) : amplitude(a), phase(p) {}
+
+    model(rng_type *rng) {
```

Comment:
This model seems nice, and is probably going to see a lot more useage. The test only uses the first constructor though, and I think the second constructor is likely going to be superseded by the following useage pattern:
```cpp
double amplitude = GENERATE(take(10, random(1, 0.5))); // 10 different amplitudes
double phase = GENERATE(take(10, random(0, M_PI / 4))); // 10 different phases
model random_model(amplitude, phase)
```
This would run the tests with 10*10 randomly generated phases and amplitudes.

So I think this rng constructor should be left out.

Reply:
> The test only uses the first constructor though

No - that is wrong; the setup in the loop:
```C++
 for (int iens = 0; iens < ens_size; iens++)
        ens.emplace_back(rng);
```